test/conversion_test.go

```diff
@@ -0,0 +1,794 @@
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  steps:
+  - name: step
+    image: gcr.io/google.com/cloudsdktool/cloud-sdk:alpine
+    command: ['/bin/bash']
+    args: ['-c', 'gcloud auth activate-service-account --key-file /var/secret/bucket-secret/bucket-secret-key']
+    workingDir: /dir
+    env:
+    - name: MY_VAR1
+      value: foo
+    resources:
+      inputs:
+      - name: workspace
+        resource: source-repo
+      outputs:
+      - name: workspace
+        resource: source-repo
+    volumeMounts:
+      - name: messages
+        mountPath: /messages
+    imagePullPolicy: IfNotPresent
+    securityContext: 
+      runAsUser: 2000
+    timeout: 60s
+    secret:
+      secretName: test-ssh-credentials
+    onError: continue
+  stepTemplate:
+    image: gcr.io/google.com/cloudsdktool/cloud-sdk:alpine
+    command: ['/bin/bash']
+    env:
+    - name: QUX
+      value: original
+    args: ['-c', 'gcloud auth activate-service-account --key-file /var/secret/bucket-secret/bucket-secret-key']
+    workingDir: /dir
+    env:
+    - name: MY_VAR1
+      value: foo
+    resources:
+      inputs:
+      - name: workspaces
+        resource: source-repo
+      outputs:
+      - name: workspace
+        resource: source-repo
+    volumeMounts:
+    - name: messages
+      mountPath: /messages
+    imagePullPolicy: IfNotPresent
+    securityContext: 
+        runAsUser: 2000
+  sidecars:
+  - name: server
+    image: alpine/git:v2.26.2
+    command: ['/bin/bash']
+    args: ['-c', 'gcloud auth activate-service-account --key-file /var/secret/bucket-secret/bucket-secret-key']
+    workingDir: /dir
+    env:
+    - name: MY_VAR1
+      value: foo
+    resources:
+      inputs:
+      - name: workspace
+        resource: source-repo
+      outputs:
+      - name: workspace
+        resource: source-repo
+    volumeMounts:
+    - name: messages
+      mountPath: /messages
+    readinessProbe:
+      periodSeconds: 1
+    securityContext:
+      runAsUser: 0
+    volumeMounts:
+    - name: messages
+      mountPath: /messages
+    script: echo test
+  volumes:
+  - name: messages
+    emptyDir: {}
+  params:
+  - name: PARAM
+    description: param des
+    type: string
+    default: "1"
+  workspaces:
+  - name: workspace
+    description: description
+    mountPath: /foo
+    readOnly: true
+    optional: true
+  resources:
+    inputs:
+      - name: git-repo
+        type: git
+        description: "The input is code from a git repository"
+        optional: true
+    outputs:
+      - name: optionalimage
+        type: image
+        description: "The output is a Docker image"
+        optional: true
+`
+
+	v1TaskYaml = `
+metadata:
+  name: %s
+  namespace: %s
+  annotations: {
+    tekton.dev/v1beta1Resources: '{"inputs":[{"name":"git-repo","type":"git","description":"The input is code from a git repository","optional":true}],"outputs":[{"name":"optionalimage","type":"image","description":"The output is a Docker image","optional":true}]}'
```

Comment:
beta resources?

Reply:
as in https://github.com/tektoncd/pipeline/pull/5579#discussion_r1028441687